packages/react-button/src/components/SplitButton/useSplitButton.ts

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import { makeMergePropsCompat, resolveShorthandProps, useMergedRefs } from '@fluentui/react-utilities';
+import type { SplitButtonProps, SplitButtonShorthandPropsCompat, SplitButtonState } from './SplitButton.types';
+
+export const splitButtonShorthandProps: SplitButtonShorthandPropsCompat[] = ['button', 'menuButton'];
+
+// eslint-disable-next-line deprecation/deprecation
+const mergeProps = makeMergePropsCompat({ deepMerge: splitButtonShorthandProps });
+
+/**
+ * Given user props, defines default props for the SplitButton and returns processed state.
+ * @param props - User provided props to the SplitButton component.
+ * @param ref - User provided ref to be passed to the SplitButton component.
+ */
+export const useSplitButton = (
+  props: SplitButtonProps,
+  ref: React.Ref<HTMLElement>,
+  defaultProps?: SplitButtonProps,
+): SplitButtonState => {
+  const resolvedShorthandProps = resolveShorthandProps(props, splitButtonShorthandProps);
+  const { className } = props;
+  const {
+    button,
+    buttonRef,
+    circular,
+    disabled,
+    disabledFocusable,
+    menuButton,
+    menuButtonRef,
+    outline,
+    primary,
+    size = 'medium',
+    subtle,
+    transparent,
+    ...userProps
+  } = resolvedShorthandProps;
+
+  const buttonInternalRef = React.useRef<HTMLElement | undefined>(null);
```

Comment:
Nit: I see this in other places.  I find it confusing to say the type is potentially undefined and then to init to null.  You could not pass anything and it would init to undefined, or you could have it be <HTMLElement | null>.